api/v1alpha1/volumesnapshotrestore_types.go

```diff
@@ -45,19 +47,31 @@ type VolumeSnapshotRestoreStatus struct {
 	StartTimestamp *metav1.Time `json:"startTimestamp,omitempty"`
 	// CompletionTimestamp records the time a volumesnapshotrestore reached a terminal state.
 	CompletionTimestamp *metav1.Time `json:"completionTimestamp,omitempty"`
-	// Includes information pertaining to Volsync ReplicationDestination CR
+	// Includes information pertaining to VolSync ReplicationDestination CR
 	ReplicationDestinationData ReplicationDestinationData `json:"replicationDestinationData,omitempty"`
 }
 
 type VSBRef struct {
 	// Includes backed up PVC name and size
-	BackedUpPVCData PVCData `json:"sourcePVCData,omitempty"`
+	BackedUpPVCData PVCData `json:"sourcePVCData,omitempty"` // this is from here api/v1alpha1/volumesnapshotbackup_types.go line 72, how it got imported?
 	// Includes restic repository path
 	ResticRepository string `json:"resticrepository,omitempty"`
 	// name of the VolumeSnapshotClass
 	VolumeSnapshotClassName string `json:"volumeSnapshotClassName,omitempty"`
 }
 
+type ReplicationDestinationData struct {
+	// name of the ReplicationDestination associated with the volumesnapshotrestore
+	Name string `json:"name,omitempty"`
+	// StartTimestamp records the time a ReplicationDestination was started.
+	// +optional
+	StartTimestamp *metav1.Time `json:"startTimestamp,omitempty"`
+	// CompletionTimestamp records the time a ReplicationDestination reached a terminal state.
+	// +optional
+	CompletionTimestamp *metav1.Time `json:"completionTimestamp,omitempty"`
+}
+
+// there is a way to extend types to avoid duplication?
```

Comment:
Here I wanted to define a type with the keys of this enum, except the first, then extend it to create one for backup and restore, with the different key. Is this possible in Go?